resources/js/Pages/Collections/Index.tsx

```diff
@@ -31,22 +34,45 @@ const CollectionsIndex = ({
 
     const { props } = usePage();
 
-    const [chain, setChain] = useState<ChainFilter | undefined>(filters.chain ?? undefined);
+    const [currentFilters, setCurrentFilters] = useState<Filters>(filters);
 
     const isFirstRender = useIsFirstRender();
 
     useEffect(() => {
         if (isFirstRender) return;
 
-        router.get(
-            route("collections"),
-            { chain },
-            {
-                preserveScroll: true,
-                preserveState: true,
-            },
-        );
-    }, [chain]);
+        const keys = Object.keys(currentFilters) as Array<keyof Filters>;
+
+        const query: RequestPayload = {};
+
+        for (const key of keys) {
+            const value = currentFilters[key];
+
+            if (value === null) continue;
+
+            query[key] = value;
+        }
+
+        router.get(route("collections"), query, {
+            only: ["collections", "activeSort", "filters"],
```

Comment:
Looks good to me. Only thing is using `only` here will not load data lazily as backend implementation doesn't have `closures` in place so we could omit this property. 

Beside that we can use `router.reload()` over `router.get()` as it will use the current URL and make this call a little bit simpler.

Ref here: https://inertiajs.com/partial-reloads